crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -92,6 +92,10 @@ pub struct Resolver<'a> {
     /// that are captured. We do this in order to create the hidden environment
     /// parameter for the lambda function.
     lambda_stack: Vec<LambdaContext>,
+
+    /// UnresolvedTypes currently don't have Spans, so if we hit an error while resolving a type
+    /// the best we can do is show the expression where the type is used
+    current_expr_span: Option<Span>,
```

Comment:
If we move this check to where the parser parses a closure type, we'll have access to the span there

Reply:
In the parser we can't tell apart NamedGenerics from struct names/type aliases.